jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/DefaultServletTest.java

```diff
@@ -471,7 +529,7 @@ public static Stream<Arguments> contextBreakoutScenarios()
             );
 
             // A Raw Question mark in the prefix can be interpreted as a query section
-            if (prefix.contains("?"))
+            if (prefix.contains("?") || prefix.contains(";"))
```

Comment:
The comment above now doesn't match the code - why are we now including semicolon?

Reply:
The tests using this methodSource were disabled in ee9.

In ee10, this same testcase is enabled.
The changes in HttpURI make some paths interpret differently here.

In ee10, we have ...

https://github.com/jetty/jetty.project/blob/745cba4a6a8f1390b752c91a699fad6d7fe115a5/jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletTest.java#L864-L865

Basically, paths like `/context/dir;/../index.html` are interpreted in Jetty 12 as `/context/dir/../index.html`
The change in the ee10 `DefaultServletTest` addresses this new reality of HttpURI, so that same change is now applied to the recently enabled tests in ee9 as well.

Reply:
OK well maybe we should put a javadoc comment in there to that effect?